planner/core/optimizer.go

```diff
@@ -278,6 +280,22 @@ func DoOptimize(ctx context.Context, sctx sessionctx.Context, flag uint64, logic
 		return nil, 0, err
 	}
 	finalPlan := postOptimize(sctx, physical)
+
+	stmtCtx := sctx.GetSessionVars().StmtCtx
+	if stmtCtx.EnableOptimizerCETrace {
+		stmtCtx.OptimizerCETrace = tracing.DedupCETrace(stmtCtx.OptimizerCETrace)
+		traceRecords := stmtCtx.OptimizerCETrace
+		is := sctx.GetInfoSchema().(infoschema.InfoSchema)
+		for _, rec := range traceRecords {
+			tbl, ok := is.TableByID(rec.TableID)
+			if !ok {
+				logutil.BgLogger().Warn("[OptimizerTrace] Failed to find table in infoschema",
+					zap.Int64("table id", rec.TableID))
+			}
+			rec.TableName = tbl.Meta().Name.O
+		}
+	}
+
```

Comment:
How about creating a new separate function for this logic instead of putting it in `DoOptimize` directly?

Reply:
Updated.